python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -287,27 +305,87 @@ async def begin_start(self, data):
             # set callback to monitor m2 valve state from now on...
             self.pneumatics.evt_m2State.callback = self.pneumatics_m2s_callback
 
+        if self.atspectrograph_summary_state is None:
+            disperser_data = None
+            filter_data = None
+
+            try:
+                self.atspectrograph_summary_state = (
+                    await self.atspectrograph.evt_summaryState.aget(timeout=self.fast_timeout)).summaryState
+
+            except asyncio.TimeoutError:
+                self.log.warning("Could not get summary state from ATSpectrograph.")
+
+            try:
+                disperser_data = await self.atspectrograph.evt_reportedDisperserPosition.aget(
+                    timeout=self.fast_timeout
+                )
+            except asyncio.TimeoutError:
+                self.log.warning("Could not get reportedDisperserPosition from ATSpectrograph.")
+
+            try:
+                filter_data = await self.atspectrograph.evt_reportedFilterPosition.aget(
+                    timeout=self.fast_timeout
+                )
+            except asyncio.TimeoutError:
+                self.log.warning("Could not get reportedFilterPosition from ATSpectrograph.")
+
+        # add focus offsets from spectrograph - this is the first time these offsets are determined
+        # these offsets get applied in correction_loop method
+        if filter_data is not None:
+            # self.model.add_offset("z", filter_data.focusOffset)
+            self.focus_offset_per_category['filterOffset'] = filter_data.focusOffset
+            self.focus_offset_yet_to_be_applied += filter_data.focusOffset
+        if disperser_data is not None:
+            # self.model.add_offset("z", disperser_data.focusOffset)
+            self.focus_offset_per_category['disperserOffset'] = disperser_data.focusOffset
+            self.focus_offset_yet_to_be_applied += disperser_data.focusOffset
+
+        self.focus_offset_per_category['totalFocusCorrectionOffset'] = self.model.offset['z']
+        self.focus_offset_per_category['cumulativeUserOffset'] = 0.0
+
+        # Add required callbacks for ATSpectrograph summary_state/filter/disperser events
```

Comment:
Add the callback as soon as you run the `aget` otherwise you can get things happening in between and loose data.